lib/logstash/filters/syslog_pri.rb

```diff
@@ -105,11 +105,20 @@ def parse_pri(event)
 
     # Add human-readable names after parsing severity and facility from PRI
     if @use_labels
-      facility_label = @facility_labels[facility_code]
+      facility_label = retrieve_with_range_check(@facility_labels, facility_code, 'facility_code')
       event.set(@facility_label_key, facility_label) if facility_label
 
-      severity_label = @severity_labels[severity_code]
+      severity_label = retrieve_with_range_check(@severity_labels, severity_code, 'severity_code')
       event.set(@severity_label_key, severity_label) if severity_label
     end
   end # def parse_pri
+
+  def retrieve_with_range_check(array, index, property_name)
+    if index > Java::JavaLang::Long::MAX_VALUE
+      logger.warn("Extracted #{property_name} too big", property_name.to_sym => index, :max_value => Java::JavaLang::Long::MAX_VALUE)
+      nil
+    else
```

Comment:
`facility_labels` and `severity_labels` arrays have limited (13 & 8) elements. I would prefer to precisely tell if requested index isn't in array size range. Does it make sense?
```suggestion
    if index >= array.size
      level_name = property_name == 'facility_code'? 'Facility' : 'Severity'
      logger.warn("#{level_name} labels do not contain value for #{index} index.")
      nil
    else
```

Reply:
Today that log is not printed, when an index doesn't fall into the array it's not translated to label, without any logging.
Moving to the `log.warn` for each index that fall outside the array could generate a lot of noisy logs I think.